SignalsIntegrationTests/SignalsIntegrationTests/BasicSignalTests.cs

```diff
@@ -172,7 +172,36 @@ public void MissingValuePolicyCanBeSetForSignal()
             Assert.AreEqual(Quality.Fair, specificMissingValuePolicy.Quality);
         }
 
-        // TODO removing?
+        [TestMethod]
+        public void WhenDeletingNonExistentSignal_Throws()
+        {
+            Assertions.AssertThrows(() => client.Delete(0));
+        }
+
+        [TestMethod]
+        public void WhenDeletingExistingSignal_SignalDisappears()
+        {
+            var signalId = AddNewIntegerSignal(granularity: Granularity.Year).Id.Value;
+
+            client.Delete(signalId);
+
+            Assertions.AssertThrows(() => client.GetById(signalId));
+        }
+
+        [TestMethod]
+        public void WhenDeletingExistingSignalWithData_BothSignalAndDataDisappear()
+        {
+            var signalId = AddNewIntegerSignal(granularity: Granularity.Year).Id.Value;
+
+            var ts = new DateTime(2000, 1, 1);
+
+            client.SetData(signalId, new[] { new Dto.Datum() { Quality = Dto.Quality.Good, Timestamp = ts, Value = 0 } });
+
+            client.Delete(signalId);
+
+            Assertions.AssertThrows(() => client.GetData(signalId, ts, ts.AddYears(1)));
```

Comment:
I'll add it to task definitions.

2016-07-05 17:26 GMT+02:00 Konrad Grochowski <notifications@github.com>:

> In SignalsIntegrationTests/SignalsIntegrationTests/BasicSignalTests.cs
> <https://github.com/bartoszkp/TDDEvaluation/pull/32#discussion_r69583305>:
>
> > +
> > +            Assertions.AssertThrows(() => client.GetById(signalId));
> > +        }
> > +
> > +        [TestMethod]
> > +        public void WhenDeletingExistingSignalWithData_BothSignalAndDataDisappear()
> > +        {
> > +            var signalId = AddNewIntegerSignal(granularity: Granularity.Year).Id.Value;
> > +
> > +            var ts = new DateTime(2000, 1, 1);
> > +
> > +            client.SetData(signalId, new[] { new Dto.Datum() { Quality = Dto.Quality.Good, Timestamp = ts, Value = 0 } });
> > +
> > +            client.Delete(signalId);
> > +
> > +            Assertions.AssertThrows(() => client.GetData(signalId, ts, ts.AddYears(1)));
>
> I do agree with the rule, question is: is this rule somewhere explicit
> provided to tests subjects?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bartoszkp/TDDEvaluation/pull/32/files/cd9420c3d4c3fb0171ad316128e6e1182ffed053#r69583305>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJZVnk4_EEa6adItSMSHrdCYcecJn1V8ks5qSneqgaJpZM4JEsDC>
> .
>